dev-tools/clickana/src/clickana.rs

```diff
@@ -0,0 +1,486 @@
+        };
+
+        let Some(max_value_bytes) =
+            values.iter().max_by(|a, b| a.partial_cmp(b).unwrap())
+        else {
+            bail!("no values have been retrieved")
+        };
+
+        // The result of these calculations will not be precise, but it doesn't matter
+        // since we just want an estimate for the labels
+        let min_value_gib = min_value_bytes.floor() as u64 / GIBIBYTE_U64;
+        let max_value_gib = max_value_bytes.ceil() as u64 / GIBIBYTE_U64;
+        let avg_value_gib = (((min_value_bytes + max_value_bytes) / 2.0).round()
+            as u64)
+            / GIBIBYTE_U64;
+
+        // In case there is very little variance in the y axis, we will be adding some
+        // buffer to the bounds and labels so we don't end up with repeated labels or
+        // straight lines too close to the upper bounds.
+        let upper_bound_value = max_value_bytes + GIBIBYTE_F64;
+        let upper_label_value = max_value_gib + 1;
+        let lower_bound_value = if min_value_bytes < &GIBIBYTE_F64 {
+            0.0
+        } else {
+            min_value_bytes - GIBIBYTE_F64
+        };
+        let lower_label_value =
+            if min_value_gib < 1 { 0 } else { min_value_gib - 1 };
+
+        // These timestamps will be used to calculate maximum and minimum values in order
+        // to create labels and set bounds for the X axis. As above, some of these conversions
+        // may lose precision, but it's OK as these values are only used to make sure the
+        // datapoints fit within the graph nicely.
+        let timestamps: Vec<i64> = raw_data
+            .iter()
+            .map(|ts| {
+                ts.time.trim_matches('"').parse::<i64>().unwrap_or_else(|_| {
+                    panic!("could not parse timestamp {} into i64", ts.time)
+                })
+            })
+            .collect();
+
+        let Some(start_time) = timestamps.iter().min() else {
+            bail!("failed to retrieve start time, timestamp list is empty")
+        };
+        let Some(end_time) = timestamps.iter().max() else {
+            bail!("failed to retrieve end time, timestamp list is empty")
+        };
+        let avg_time = (*start_time + *end_time) / 2;
+        let Some(start_time_utc) = DateTime::from_timestamp(*start_time, 0)
+        else {
+            bail!(
+                "failed to convert timestamp to UTC date and time;
+        timestamp = {}",
+                start_time
+            )
+        };
+        let Some(end_time_utc) = DateTime::from_timestamp(*end_time, 0) else {
+            bail!(
+                "failed to convert timestamp to UTC date and time;
+        timestamp = {}",
+                end_time
+            )
+        };
+        let Some(avg_time_utc) = DateTime::from_timestamp(avg_time, 0) else {
+            bail!(
+                "failed to convert timestamp to UTC date and time;
+        timestamp = {}",
+                avg_time
+            )
+        };
```

Comment:
Kinda on the fence on whether I should only display the time without the date. The label ends up way too long and makes the chart look kinda bad on the left side. On the other hand, times without dates will be problematic when visualizing data from more than one day.